internal/model/todos.go

```diff
@@ -19,81 +19,119 @@ type TodoList struct {
+		return err
+	}
+	sql := `INSERT INTO lists(name) VALUES($1) RETURNING id;`
+	err := db.QueryRow(sql, tl.Name).Scan(&tl.ID)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	for rows.Next() {
-		todo := &Todo{}
-		err := rows.Scan(&todo.ID, &todo.ListID, &todo.Todo, &todo.Done)
+	for _, t := range tl.Todos {
+		t.ListID = tl.ID
+		err := t.Save()
 		if err != nil {
-			return nil, err
+			return err
 		}
-		list.Todos = append(list.Todos, todo)
 	}
-	return list, nil
+
+	return nil
 }
 
-// CreateTodos inserts todos into the todos table
-func CreateTodos(name string, todos []string) ([]*Todo, error) {
-	var newTodos []*Todo
-	listID, err := createList(name)
+// TodoByID fetches a single todo given a listID and todoID
+func TodoByID(listID, todoID string) (*Todo, error) {
+	t := &Todo{}
+	sql := `SELECT id, list_id, todo, done FROM todos WHERE list_id = $1 AND id = $2 ORDER BY created_at ASC;`
+	err := db.QueryRow(sql, listID, todoID).Scan(&t.ID, &t.ListID, &t.Todo, &t.Done)
+	if err != nil {
+		return nil, err
+	}
+
+	return t, nil
+}
 
+// TodosByListID fetches the todos for a given list ID
+func TodosByListID(id string) ([]*Todo, error) {
+	sql := `SELECT id, list_id, todo, done FROM todos WHERE list_id = $1 ORDER BY created_at ASC;`
+	rows, err := db.Query(sql, id)
 	if err != nil {
 		return nil, err
 	}
 
-	sqlStatement := `INSERT INTO todos(list_id, todo) VALUES($1, $2)
-		RETURNING id, list_id, todo, done`
-	for _, todo := range todos {
-		var createdTodo = &Todo{}
-		err := db.QueryRow(sqlStatement, listID, todo).Scan(&createdTodo.ID, &createdTodo.ListID, &createdTodo.Todo, &createdTodo.Done)
+	var todos []*Todo
+	for rows.Next() {
+		todo := &Todo{}
+		err := rows.Scan(&todo.ID, &todo.ListID, &todo.Todo, &todo.Done)
 		if err != nil {
 			return nil, err
 		}
-		if createdTodo.ID == "" {
-			return nil, errors.New("created todo ID empty")
-		}
-		newTodos = append(newTodos, createdTodo)
+		todos = append(todos, todo)
 	}
-	return newTodos, nil
+
+	return todos, nil
 }
 
-// createList creates a list and returns the UUID
-func createList(name string) (string, error) {
-	var id string
-	sqlStatement := `INSERT INTO lists(name) VALUES($1) RETURNING id;`
-	err := db.QueryRow(sqlStatement, name).Scan(&id)
-	if err != nil {
-		return "", err
+func (t *Todo) validate() error {
+	if t.ListID == "" {
+		return errors.New("ListID is empty, must be set")
 	}
-	return id, nil
+	if t.Todo == "" {
+		return errors.New("Todo is empty, must be set")
+	}
+	return nil
 }
 
-// MarkTodoAsDone mark todo as done or undone depending on the value passed in
-func MarkTodoAsDone(listID string, todoID string, done bool) (*Todo, error) {
-	todo := &Todo{}
+// Save saves an individual todo to the database
+func (t *Todo) Save() error {
+	if err := t.validate(); err != nil {
+		return err
+	}
+
+	sql := `INSERT INTO todos(list_id, todo) VALUES($1, $2) RETURNING id`
+	err := db.QueryRow(sql, t.ListID, t.Todo).Scan(&t.ID)
+	if err != nil {
+		return err
+	}
```

Comment:
We can call `Save()` on an individual Todo now keeping the API consistent and the SQL living alongside the model/struct itself.